pkg/block/fetcher.go

```diff
@@ -735,21 +764,26 @@ func NewConsistencyDelayMetaFilter(
 
 // Filter filters out blocks that filters blocks that have are created before a specified consistency delay.
 func (f *ConsistencyDelayMetaFilter) Filter(ctx context.Context, metas map[ulid.ULID]*metadata.Meta, synced *extprom.TxGaugeVec) error {
-	// If consistencyDelay is 0, skip filtering.
 	if f.consistencyDelay == 0 {
 		return nil
 	}
 
 	for id, meta := range metas {
-		metaFile := path.Join(id.String(), MetaFilename)
-		attributes, err := f.bkt.Attributes(ctx, metaFile)
-		if err != nil {
-			return errors.Wrapf(err, "get object attributes of %s", metaFile)
+		lastModifiedTime := meta.Thanos.LastModified
+		// Will this happen?
+		if lastModifiedTime.IsZero() {
```

Comment:
Umm, this seems unlikely to happen, but I am not sure if it is good to remove it.

Reply:
I would simplify this, as follow:
- Remove this `if` branch, so that the filter doesn't need the objstore client
- If last modified is zero (for whatever reason), then we can just fallback to the old logic, which is reading the timestamp from the block ID (which is the timestamp when the block creation started)

Reply:
Do we still want to keep the `timestamp based block ID` way? I thought we want to fully change to use `lastModifiedTime` based on the comment

``` 
// TODO(bwplotka): Check consistency delay based on file upload / modification time instead of ULID.
```

If it is this case, then we don't need to have `lastModifiedTime` in the json file and the logic would be much easier! Thanks.